http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -1288,20 +1288,22 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         dbNames <-
           dbNamePrefixes
             .traverse(databaseNamePrefix =>
-              for {
-                databaseName <-
-                  createDatabaseInWsm(app,
-                                      workspaceId,
-                                      namespace,
-                                      databaseNamePrefix,
-                                      wsmApi,
-                                      Option(createIdentityResponse.getResourceId)
-                  )
-              } yield databaseNamePrefix -> databaseName
+              createDatabaseInWsm(app,
+                                  workspaceId,
+                                  namespace,
+                                  databaseNamePrefix,
+                                  wsmApi,
+                                  Option(createIdentityResponse.getResourceId)
+              )
             )
-            .map(list => list.toMap)
 
-      } yield (Some(ServiceAccountName(identityName)), Some(dbNames))
+        typedDbNames = app.appType match {
+          case AppType.Wds          => Some(WdsDatabaseNames(dbNames.head))
+          case AppType.WorkflowsApp => Some(WorkflowsAppDatabaseNames(dbNames.head, dbNames.apply(1)))
```

Comment:
Probably not a problem here... but making assumptions about how long a list will be can sometimes backfire. I wonder whether any kind of validation on the size makes sense? Or is that overly paranoid?